src/ext/resources/legacygraphqlapi/schema.graphqls

```diff
@@ -1376,6 +1376,15 @@ type Leg {
     dropOffBookingInfo: BookingInfo
 
     pickupBookingInfo: BookingInfo
+
+    """
+    Future legs with same origin and destination stops or stations
+    """
+    nextLegs(
+    """The number of alternative legs searched. If fewer than the requested number are found, then only the found legs are returned"""
+    numberOfLegs: Int!,
+     """Transportation modes for which all stops in the parent station are used as origin stops. For modes not listed, only the exact origin stop of the leg is considered"""
+     modesWithParentStation: [Mode]): [Leg]
```

Comment:
```suggestion
    nextLegs(
        """
        The number of alternative legs searched. If fewer than the requested number are found,
        then only the found legs are returned.
        """
        numberOfLegs: Int!

        """
        Transportation modes for which all stops in the parent station are used as possible origin stops
        for the next legs. For modes not listed, only the exact origin stop of the leg is considered.
        """
        modesWithParentStation: [Mode!]): [Leg!]
```

Reply:
This suggestion has the following changes:
* Indentation so it's clearer that the arguments belong to the field as otherwise it can look like they are fields rather than arguments and split some comments on multiple lines
* Made minor clarification for what the origin stops mean
* Made Mode and Leg non null

There are still a couple of things that probably should be changed:
* The `modesWithParentStation` now only affects the origin stop. However, I think we should use the same policy for the destination stop (in our use case) as, for example, the trains leaving from different tracks often arrive at different tracks at the destination. Although, I can see that someone might want to have different behaviour for these so I think we could split this parameter as two, one for the origin and one for the destination something like `originModesWithParentStation` and  `destinationModesWithParentStation`.
* I think we could make a new enum in the schema that could be called `TransitMode` which doesn't have BICYCLE, CAR, FLEXIBLE, LEG_SWITCH, SCOOTER, CARPOOL, TAXI, TRANSIT, WALK. I don't know if it should include FLEX  or not, @hannesj probably has a better idea if it would be possible to have alternative legs for flex.

Reply:
After thinking about this for a moment, I think actually the new enum probably should be called something more specific like `AlternativeLegsMode` so it's more flexible to fit this use case as opposite to an enum called `TransitMode` that could potentially be used for something else, and we don't really need to include modes here that don't fit this use case but fit some other "transit mode" use case.

Reply:
These have now been implemented

Reply:
It seems that I forgot to push - now the changes should be visible

Reply:
Instead of using the `Mode` enum in the schema, add a `TransitMode` enum to the schema and use it for the originModesWithParentStation and destinationModesWithParentStation. I think in addition you have to define the new TransitMode to be mapped to string here https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/generated/graphql-codegen.yml#L58

Reply:
Made those changes